test/Quaternion.jl

```diff
@@ -171,6 +202,19 @@ end
         end
     end
 
+    @testset "inv does not under/overflow" begin
+        @test isequal(inv(quat(1e300, 0.0, 0.0, 0.0)), quat(1e-300, -0.0, -0.0, -0.0))
+        @test isequal(inv(quat(0.0, 1e300, 0.0, 0.0)), quat(0.0, -1e-300, -0.0, -0.0))
+        @test isequal(inv(quat(0.0, 0.0, 1e300, 0.0)), quat(0.0, -0.0, -1e-300, -0.0))
+        @test isequal(inv(quat(0.0, 0.0, 0.0, 1e300)), quat(0.0, -0.0, -0.0, -1e-300))
+        @test inv(quat(1e-300, 0.0, 0.0, 0.0)) ≈ quat(1e300, -0.0, -0.0, -0.0)
+        @test inv(quat(0.0, 1e-300, 0.0, 0.0)) ≈ quat(0.0, -1e300, -0.0, -0.0)
+        @test inv(quat(0.0, 0.0, 1e-300, 0.0)) ≈ quat(0.0, -0.0, -1e300, -0.0)
+        @test inv(quat(0.0, 0.0, 0.0, 1e-300)) ≈ quat(0.0, -0.0, -0.0, -1e300)
+        @test isequal(inv(quat(-Inf, 1, -2, 3)), quat(-0.0, -0.0, 0.0, -0.0))
+        @test isequal(inv(quat(1, -2, Inf, 3)), quat(0.0, 0.0, -0.0, -0.0))
```

Comment:
Yeah, the `isequal` checks that the sign on the 0s is correct.

Reply:
Ah, I see. I misunderstood because:

* The title for `@testset` is "inv does not under/overflow".
* Both `isequal` and `≈` are used.

The reason for `≈` is just for `inv(1e-300) ≠ 1e300`, so replacing them with `x, y=inv(x)` is better here?
```suggestion
        x = 1e300
        y = inv(x)
        @test isequal(inv(quat(x, 0.0, 0.0, 0.0)), quat(y, -0.0, -0.0, -0.0))
        @test isequal(inv(quat(0.0, x, 0.0, 0.0)), quat(0.0, -y, -0.0, -0.0))
        @test isequal(inv(quat(0.0, 0.0, x, 0.0)), quat(0.0, -0.0, -y, -0.0))
        @test isequal(inv(quat(0.0, 0.0, 0.0, x)), quat(0.0, -0.0, -0.0, -y))
        @test isequal(inv(quat(y, 0.0, 0.0, 0.0)), quat(x, -0.0, -0.0, -0.0))
        @test isequal(inv(quat(0.0, y, 0.0, 0.0)), quat(0.0, -x, -0.0, -0.0))
        @test isequal(inv(quat(0.0, 0.0, y, 0.0)), quat(0.0, -0.0, -x, -0.0))
        @test isequal(inv(quat(0.0, 0.0, 0.0, y)), quat(0.0, -0.0, -0.0, -x))
        @test isequal(inv(quat(-Inf, 1, -2, 3)), quat(-0.0, -0.0, 0.0, -0.0))
        @test isequal(inv(quat(1, -2, Inf, 3)), quat(0.0, 0.0, -0.0, -0.0))
```

Reply:
Agreed, that should be better